service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt

```diff
@@ -177,6 +179,17 @@ class AsyncCaching(
         return TxUpdatedItems(addrs, markers)
     }
 
+    private fun saveEvents(txId: EntityID<Int>, tx: ServiceOuterClass.GetTxResponse, msg: com.google.protobuf.Any, msgId: String, msgType: String) = transaction {
+        tx.txResponse.logsList.forEach { log ->
```

Comment:
You dont want to save all events in the tx against only one msg. you need to match the logsList index to the msgList index.